internal/db/schema/migrations/oss/postgres/59/01_ldap.up.sql

```diff
@@ -0,0 +1,350 @@
+  for each row execute procedure immutable_columns('public_id', 'scope_id', 'create_time');
+
+create trigger default_create_time_column before insert on auth_ldap_method
+  for each row execute procedure default_create_time();
+
+create trigger update_version_column after update on auth_ldap_method
+  for each row execute procedure update_version_column();
+
+-- auth_ldap_url entries are LDAP URLs that specify an LDAP servers to connect
+-- to. Examples: ldap://ldap.myorg.com, ldaps://ldap.myorg.com:636. There must
+-- be at least one and if there's more than one URL configured for an auth
+-- method, the directories will be tried in connection_priority order if there
+-- are errors during the connection process.  The URL scheme must be either ldap
+-- or ldaps. The port is optional.If no port is specified, then a default of 389
+-- is used for ldap and a default of 689 is used for ldaps. (see rfc4516 for
+-- more information about LDAP URLs)  
+--
+-- Updates will be implemented as a delete + insert with the auth_ldap_method
+-- being used as the root aggregate for auth_ldap_url updates. 
+create table auth_ldap_url ( 
+  create_time wt_timestamp,
+  ldap_method_id wt_public_id not null  
+    constraint auth_ldap_method_fkey
+      references auth_ldap_method(public_id)
+      on delete cascade
+      on update cascade,
+  url wt_url not null,
+  connection_priority int not null
+    constraint connection_priority_less_than_zero
+      check(connection_priority >= 0),
+  primary key(ldap_method_id, connection_priority)
+);
+comment on table auth_ldap_url is
+'auth_ldap_url entries specify a connection URL an LDAP';
+
+create or replace function auth_ldap_url_parent_children() returns trigger 
+as $$
+declare 
+  n integer;
+begin
+  if tg_op = 'INSERT' or tg_op = 'UPDATE' then 
+    select into n count(*) from auth_ldap_url where ldap_method_id = new.ldap_method_id;
+    if n < 1 then 
+      raise exception 'During % of auth_ldap_url: auth_ldap_method id=% must have at least one url, not %',tg_op,new.ldap_method_id,n;
+    end if;
+  end if;
+  if tg_op = 'UPDATE' or tg_op = 'DELETE' then 
+      select into n count(*) from auth_ldap_url where ldap_method_id = old.ldap_method_id;
+      if n < 1 then 
+        raise exception 'During % of auth_ldap_url: auth_ldap_method id=% must have at least one url, not %',tg_op,old.ldap_method_id,n;
+      end if;
+  end if;
+
+  return null;
+end;
+$$ language plpgsql;
+comment on function auth_ldap_url_parent_children() is 
+'function used on auth_ldap_url after insert/update/delete initially deferred to ensure each '
+'auth_ldap_method has at least one auth_ldap_url';
+
+create constraint trigger auth_ldap_url_children_per_parent_tg
+  after insert or update or delete on auth_ldap_url deferrable initially deferred 
+  for each row execute procedure auth_ldap_url_parent_children();
+
+create or replace function auth_ldap_method_children() returns trigger as $$
```

Comment:
```suggestion
create function auth_ldap_method_children() returns trigger
as $$
```
* Same comment as above regarding `or replace`
* We put the `as $$` on a new line

Reply:
ty!